zappa/core.py

```diff
@@ -1198,29 +1202,34 @@ def update_lambda_configuration(    self,
         # Check if there are any remote aws lambda env vars so they don't get trashed.
         # https://github.com/Miserlou/Zappa/issues/987,  Related: https://github.com/Miserlou/Zappa/issues/765
         lambda_aws_config = self.lambda_client.get_function_configuration(FunctionName=function_name)
+        is_docker_deployment = lambda_aws_config['PackageType'] == 'Image'
         if "Environment" in lambda_aws_config:
             lambda_aws_environment_variables = lambda_aws_config["Environment"].get("Variables", {})
             # Append keys that are remote but not in settings file
             for key, value in lambda_aws_environment_variables.items():
                 if key not in aws_environment_variables:
                     aws_environment_variables[key] = value
 
-        response = self.lambda_client.update_function_configuration(
-            FunctionName=function_name,
-            Runtime=runtime,
-            Role=self.credentials_arn,
-            Handler=handler,
-            Description=description,
-            Timeout=timeout,
-            MemorySize=memory_size,
-            VpcConfig=vpc_config,
-            Environment={'Variables': aws_environment_variables},
-            KMSKeyArn=aws_kms_key_arn,
-            TracingConfig={
+        kwargs = {
+            'FunctionName': function_name,
+            'Role': self.credentials_arn,
+            'Description': description,
+            'Timeout': timeout,
+            'MemorySize': memory_size,
+            'VpcConfig':vpc_config,
+            'Environment':{'Variables': aws_environment_variables},
+            'KMSKeyArn': aws_kms_key_arn,
+            'TracingConfig': {
                 'Mode': 'Active' if self.xray_tracing else 'PassThrough'
             },
-            Layers=layers
-        )
+        }
+
+        if not is_docker_deployment:
+            kwargs['Handler'] = handler
+            kwargs['Runtime'] = runtime
+            kwargs['Layers'] = layers
+
+        response = self.lambda_client.update_function_configuration(**kwargs)
```

Comment:
Any reason for not simplifying this down to

```python
        lambda_aws_config = self.lambda_client.get_function_configuration(FunctionName=function_name)
        if "Environment" in lambda_aws_config:
            lambda_aws_environment_variables = lambda_aws_config["Environment"].get("Variables", {})
            # Append keys that are remote but not in settings file
            for key, value in lambda_aws_environment_variables.items():
                if key not in aws_environment_variables:
                    aws_environment_variables[key] = value

        kwargs = {
            'FunctionName': function_name,
            'Role': self.credentials_arn,
            'Description': description,
            'Timeout': timeout,
            'MemorySize': memory_size,
            'VpcConfig':vpc_config,
            'Environment':{'Variables': aws_environment_variables},
            'KMSKeyArn': aws_kms_key_arn,
            'TracingConfig': {
                'Mode': 'Active' if self.xray_tracing else 'PassThrough'
            },
        }

        if lambda_aws_config['PackageType'] != 'Image':
            kwargs.update({
                'Handler': handler,
                'Runtime': runtime,
                'Layers': layers,
            }

        response = self.lambda_client.update_function_configuration(**kwargs)
```

I could probably get fancy with an inline ternary dict unpacking in the original function call but I like the readability of the `kwargs` definition here, and chose to match that by passing `kwargs.update()` a dictionary so the keys all look the same.

Reply:
No particular reason, I agree this is more readable. I will update

Reply:
Updated.